webhooks/virtualmachinepublishrequest/validation/virtualmachinepublishrequest_validator.go

```diff
@@ -165,6 +173,43 @@ func (v validator) validateTargetLocation(ctx *pkgctx.WebhookRequestContext, vmp
 	return allErrs
 }
 
+// validateCreateVMGroupPublishRequestOwnership ensures a VirtualMachineGroupPublishRequest owner reference is set
+// when a VirtualMachinePublishRequest has a managed by VirtualMachineGroupPublishRequest label.
+func (v validator) validateCreateVMGroupPublishRequestOwnership(vmPub *vmopv1.VirtualMachinePublishRequest) error {
+	if !metav1.HasLabel(vmPub.ObjectMeta, vmopv1.VirtualMachinePublishRequestManagedByLabelKey) {
+		return nil
+	}
+
+	for _, ownerRef := range vmPub.OwnerReferences {
+		if ownerRef.Kind == "VirtualMachineGroupPublishRequest" &&
+			ownerRef.Name == vmPub.Labels[vmopv1.VirtualMachinePublishRequestManagedByLabelKey] {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("%q label is reserved", vmopv1.VirtualMachinePublishRequestManagedByLabelKey)
```

Comment:
It would be helpful to show what's described in the function comment. There is no `VirtualMachineGroupPublishRequest` in owner ref